src/test/ui/parser/removed-syntax-with-1.rs

```diff
@@ -7,5 +7,5 @@ fn main() {
     let a = S { foo: (), bar: () };
     let b = S { foo: () with a };
     //~^ ERROR expected one of `,`, `.`, `?`, `}`, or an operator, found `with`
-    //~| ERROR missing field `bar` in initializer of `main::S`
+    //~| ERROR missing fields `bar`, `foo` in initializer of `main::S`
```

Comment:
This feels like a regression to me. The `foo` field has been given. I understand this was done deliberately (https://github.com/rust-lang/rust/pull/57779/files#diff-da9d34ca1d0f6beee2838cf02e07345cR2720) for the `S { foo.cake(), bar: () }` situation. Do you think it would be feasible to only report the field if `parse_field` parsed a shorthand field pattern? If a `:` was specified, and the field was parsed successfully, then the field isn't "missing" per se.

Reply:
Done.